py/frequenz/client/weather/_historical_forecast_iterator.py

```diff
@@ -65,29 +65,26 @@ async def __anext__(self) -> weather_pb2.LocationForecast:
         Raises:
             StopAsyncIteration: If there are no more historical weather forecasts.
         """
-        if len(self.location_forecasts) == 0 and self.page_token == EMPTY_PAGE_TOKEN:
+        if self.page_token == EMPTY_PAGE_TOKEN:
             raise StopAsyncIteration
 
-        if self.location_forecasts is None or len(self.location_forecasts) == 0:
-            pagination_params = pagination_params_pb2.PaginationParams()
-            pagination_params.page_size = PAGE_SIZE
-            if self.page_token is not None:
-                pagination_params.page_token = self.page_token
-
-            response: Any = (
-                await self._stub.GetHistoricalWeatherForecast(  # type:ignore
-                    weather_pb2.GetHistoricalWeatherForecastRequest(
-                        locations=(location.to_pb() for location in self.locations),
-                        features=(feature.value for feature in self.features),
-                        start_ts=self.start_ts,
-                        end_ts=self.end_ts,
-                        pagination_params=pagination_params,
-                    )
-                )
+        pagination_params = pagination_params_pb2.PaginationParams()
+        pagination_params.page_size = PAGE_SIZE
+        if self.page_token is not None:
+            pagination_params.page_token = self.page_token
+
+        response: Any = await self._stub.GetHistoricalWeatherForecast(  # type:ignore
+            weather_pb2.GetHistoricalWeatherForecastRequest(
+                locations=(location.to_pb() for location in self.locations),
+                features=(feature.value for feature in self.features),
+                start_ts=self.start_ts,
+                end_ts=self.end_ts,
+                pagination_params=pagination_params,
             )
+        )
 
-            self.page_token = response.pagination_info.next_page_token
-            self.location_forecasts = response.location_forecasts
+        self.page_token = response.pagination_info.next_page_token
```

Comment:
Are we sure that this will never be `None` after this line?

Reply:
The API spec does not guarantee this. This assumption is based on the current service implementation which is subject to a change.

Reply:
So I added an additional condition checking for `None` to be able to catch this in case the service implementation changes.